puppets.js

```diff
@@ -139,10 +139,11 @@ const getDeviceScaleFactorData = async browser => {
             },
             { device: '', scaleFactor: 1 },
           );
-        },
-      );
-    return scrapeScaleFactorDataFromHIGPage();
-  });
+        });
+      return scrapeScaleFactorDataFromHIGPage();
+    },
+    { selector: constants.APPLE_HIG_SPLASH_SCR_SPECS_DATA_GRID_SELECTOR },
+  );
 
   if (scaleFactorData == null) {
```

Comment:
Could you add strict equality operator in here, please?

Reply:
Although your remark is not related to the changeset of this PR, it's a valid point and I did necessary changes as it might produce another bug. The data returned here is neither null or undefined, instead an empty array, introduced array.length check here.